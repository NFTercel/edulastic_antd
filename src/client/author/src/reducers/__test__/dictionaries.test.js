import dictionariesReducer from "../dictionaries";

import {
  RECEIVE_DICT_CURRICULUMS_REQUEST,
  RECEIVE_DICT_CURRICULUMS_SUCCESS,
  RECEIVE_DICT_CURRICULUMS_ERROR,
  RECEIVE_DICT_STANDARDS_REQUEST,
  RECEIVE_DICT_STANDARDS_SUCCESS,
  RECEIVE_DICT_STANDARDS_ERROR
} from "../../constants/actions";

describe("dictionaries", () => {
  it("should return the initial state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {}
      )
    ).toMatchSnapshot();
  });

  it("should return the receive dictionary curriculums request state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {
          type: RECEIVE_DICT_CURRICULUMS_REQUEST,
          payload: {}
        }
      )
    ).toMatchSnapshot();
  });

  it("should return the receive dictionary curriculums success state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {
          type: RECEIVE_DICT_CURRICULUMS_SUCCESS,
          payload: {
            items: {
              curriculumId: "5wrg64u51qhiuytuiogazcv",
              grades: 5,
              search: "match"
            }
          }
        }
      )
    ).toMatchSnapshot();
  });

  it("should return the receive dictionary curriculums error state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {
          type: RECEIVE_DICT_CURRICULUMS_ERROR,
          payload: {
            error: "unexpected error happened"
          }
        }
      )
    ).toMatchSnapshot();
  });

  it("should return the receive dictionary standard request state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {
          type: RECEIVE_DICT_STANDARDS_REQUEST,
          payload: {}
        }
      )
    ).toMatchSnapshot();
  });

  it("should return the receive dictionary standard success state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {
          type: RECEIVE_DICT_STANDARDS_SUCCESS,
          payload: {
            items: {
              curriculumId: 5
            }
          }
        }
      )
    ).toMatchSnapshot();
  });

  it("should return the receive dictionary standard error state", () => {
    expect(
      dictionariesReducer(
        {
          curriculums: {
            curriculums: [],
            loading: false,
            error: null
          },
          standards: {
            standards: [],
            loading: false,
            error: null
          }
        },
        {
          type: RECEIVE_DICT_STANDARDS_ERROR,
          payload: {
            error: "receive dictionary standard error happened"
          }
        }
      )
    ).toMatchSnapshot();
  });
});
