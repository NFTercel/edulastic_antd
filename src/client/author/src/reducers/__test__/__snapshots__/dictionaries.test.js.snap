// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dictionaries should return the initial state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Array [],
    "error": null,
    "loading": false,
  },
  "standards": Object {
    "error": null,
    "loading": false,
    "standards": Array [],
  },
}
`;

exports[`dictionaries should return the receive dictionary curriculums error state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Array [],
    "error": "unexpected error happened",
    "loading": false,
  },
  "standards": Object {
    "error": null,
    "loading": false,
    "standards": Array [],
  },
}
`;

exports[`dictionaries should return the receive dictionary curriculums request state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Array [],
    "error": null,
    "loading": true,
  },
  "standards": Object {
    "error": null,
    "loading": false,
    "standards": Array [],
  },
}
`;

exports[`dictionaries should return the receive dictionary curriculums success state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Object {
      "curriculumId": "5wrg64u51qhiuytuiogazcv",
      "grades": 5,
      "search": "match",
    },
    "error": null,
    "loading": false,
  },
  "standards": Object {
    "error": null,
    "loading": false,
    "standards": Array [],
  },
}
`;

exports[`dictionaries should return the receive dictionary standard error state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Array [],
    "error": null,
    "loading": false,
  },
  "standards": Object {
    "error": "receive dictionary standard error happened",
    "loading": false,
    "standards": Array [],
  },
}
`;

exports[`dictionaries should return the receive dictionary standard request state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Array [],
    "error": null,
    "loading": false,
  },
  "standards": Object {
    "error": null,
    "loading": true,
    "standards": Array [],
  },
}
`;

exports[`dictionaries should return the receive dictionary standard success state 1`] = `
Object {
  "curriculums": Object {
    "curriculums": Array [],
    "error": null,
    "loading": false,
  },
  "standards": Object {
    "error": null,
    "loading": false,
    "standards": Object {
      "curriculumId": 5,
    },
  },
}
`;
